src/System.Text.Primitives/System/Text/Encoding/Internal/ParsingTrie.cs

```diff
@@ -311,10 +300,44 @@ public static Node[] Create(byte[][] symbols)
             // counting the number of children a node has.
             symbolList.Sort();
 
+            // validate symbol consistemcy:
+            //   a) each symbol must be unique
+            //   b) a symbol cannot be a prefix of another symbol
+            //   c) symbols cannot be empty
+            for(int i=1; i<symbolList.Count; i++)
```

Comment:
nit: I forgot to comment here, same spacing.

Also, should we add a test to confirm the fix?